daemon/cmd/kube_proxy_replacement.go

```diff
@@ -312,6 +315,13 @@ func initKubeProxyReplacementOptions() (bool, error) {
 		}
 		if !option.Config.EnableHostServicesTCP && !option.Config.EnableHostServicesUDP {
 			option.Config.EnableSocketLB = false
+			option.Config.EnableSocketLBTracing = false
+		}
+		option.Config.EnableSocketLBTracing = true
+		if h := probesManager.GetHelpers("cgroup_sock_addr"); h != nil {
+			if _, found := h["bpf_perf_event_output"]; !found {
+				option.Config.EnableSocketLBTracing = false
```

Comment:
I think this should be accompanied with a warning if this overrides the user's configuration? Otherwise it will just silently turn off if the kernel support isn't there.